src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,90 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private final int[] nums;
+    private final int start;
+    private final int end;
 
     public MergeSortAction(int[] array) {
+        this(array, 0, array.length - 1);
+    }
+
+    private MergeSortAction(int[] array, int start, int end) {
+        nums = array;
+        this.start = start;
+        this.end = end;
+    }
+
+    @Override
+    protected void compute() {
+        if ((end - start + 1) <= 1) {
+            return;
+        }
+
+        int mid = (start + end) / 2;
+
+        MergeSortAction leftFork = new MergeSortAction(nums, start, mid);
+        MergeSortAction rightFork = new MergeSortAction(nums, mid + 1, end);
+
+        leftFork.fork();
+        rightFork.fork();
+        leftFork.join();
+        rightFork.join();
```

Comment:
Instead of calling `fork()` on both `leftFork` and `rightFork` and then calling `join()` on both, you can optimize the performance by calling `compute()` directly on one of the forks (e.g., `leftFork.compute()`) and only using `fork()` and `join()` on the other (e.g., `rightFork`). This reduces the overhead of task scheduling in the Fork/Join framework.